cpu/rpx0xx/periph/usbdev.c

```diff
@@ -0,0 +1,860 @@
+                                   usb_ep_dir_t dir, size_t size)
+{
+    rpx0xx_usb_ep_t *hw_ep = NULL;
+
+    DEBUG("[rpx0xx usb] Call _usbdev_new_ep(dev=..., type=%d, " 
+          "dir=%s, size=%u)\n", type, PRINT_DIR(dir), size);
+
+    /* Always return endpoint 0 for control types */
+    if (type == USB_EP_TYPE_CONTROL) {
+        hw_ep = _get_ep(0, dir);
+        hw_ep->ep.num = 0;
+    }
+    else {
+        /* Find the first unassigned ep with proper dir */
+        for (unsigned i = 1; i < USBDEV_NUM_ENDPOINTS; i++) {
+            hw_ep = _get_ep(i, dir);
+            if (hw_ep->ep.type == USB_EP_TYPE_NONE) {
+                hw_ep->ep.num = i;
+                break;
+            }
+        }
+    }
+    if (hw_ep) {
+        hw_ep->ep.dev = dev;
+        hw_ep->ep.dir = dir;
+        hw_ep->ep.type = type;
+        hw_ep->ep.len = size;
+        _ep_dpram_alloc(hw_ep, size);
+    }
+
+    return &hw_ep->ep;
+}
+
+static void _enable_ep_irq(usbdev_ep_t *ep) {
+    if (ep->num != 0) {
+        uint32_t flags = USBCTRL_DPRAM_EP1_IN_CONTROL_INTERRUPT_PER_BUFF_Msk
+                         | USBCTRL_DPRAM_EP1_IN_CONTROL_INTERRUPT_ON_STALL_Msk;
+        _ep_ctrl_reg_write(ep->num, ep->dir, flags, flags);
+    } else {
+        uint32_t flags = USBCTRL_REGS_SIE_CTRL_EP0_INT_1BUF_Msk
+                         | USBCTRL_REGS_SIE_CTRL_EP0_INT_STALL_Msk;
+        io_reg_atomic_set(&USBCTRL_REGS->SIE_CTRL, flags);
+    }
+}
+
+static void _disable_ep_irq(usbdev_ep_t *ep) {
+    if (ep->num != 0) {
+        uint32_t flags = USBCTRL_DPRAM_EP1_IN_CONTROL_INTERRUPT_PER_BUFF_Msk
+                         | USBCTRL_DPRAM_EP1_IN_CONTROL_INTERRUPT_ON_STALL_Msk;
+        _ep_ctrl_reg_write(ep->num, ep->dir, 0, flags);
+    } else {
+        uint32_t flags = USBCTRL_REGS_SIE_CTRL_EP0_INT_1BUF_Msk
+                         | USBCTRL_REGS_SIE_CTRL_EP0_INT_STALL_Msk;
+        io_reg_atomic_clear(&USBCTRL_REGS->SIE_CTRL, flags);
+    }
+}
+
+static void _usbdev_ep_init(usbdev_ep_t *ep)
+{
+    DEBUG("[rpx0xx usb] Call _usbdev_ep_init(ep.num=%d, " 
+          "ep.dir=%s)\n", ep->num, PRINT_DIR(ep->dir));
+
+    rpx0xx_usb_ep_t *hw_ep = _get_ep(ep->num, ep->dir);
+
+    hw_ep->data_buf = NULL;
+    hw_ep->data_buf_size = 0UL;
+    hw_ep->data_cnt = 0UL;
+    hw_ep->next_pid = 1U;
+}
+
+static void _ep_enable(usbdev_ep_t *ep) {
+    if (ep->num != 0) {
+        _ep_ctrl_reg_write(ep->num, ep->dir,
+                           USBCTRL_DPRAM_EP1_IN_CONTROL_ENABLE_Msk,
+                           USBCTRL_DPRAM_EP1_IN_CONTROL_ENABLE_Msk);
+    }
+}
+
+static void _ep_disable(usbdev_ep_t *ep) {
+    if (ep->num != 0) {
+        _ep_ctrl_reg_write(ep->num, ep->dir, 0,
+                           USBCTRL_DPRAM_EP1_IN_CONTROL_ENABLE_Msk);
+    }
+}
+
+static void _ep_set_stall(usbdev_ep_t *ep, usbopt_enable_t enable)
+{
+    rpx0xx_usb_ep_t *hw_ep = _get_ep(ep->num, ep->dir);
+    hw_ep->next_pid = 0u;
```

Comment:
```suggestion

```
I think this line should be remove for a correct PID management otherwise we end up always using the same PID.